lib/resty/upstream/healthcheck.lua

```diff
@@ -134,7 +138,15 @@ local function peer_fail(ctx, is_backup, id, peer)
     -- print("ctx fall: ", ctx.fall, ", peer down: ", peer.down,
           -- ", fails: ", fails)
 
-    if not peer.down and fails >= ctx.fall then
+    local u_key = gen_upstream_key(u,"all_down")
+
+    local all_down, err = dict:get(u_key)
+    if not all_down then
+        errlog("Failed to get all down flag for upstream " .. u .. ". ", err)
```

Comment:
I think we do not use upper case letter at the beginning of our error messages. Also, do we really want this to be fatal? The health check service would be down when the shm zone is running out of storage, which does not look very graceful to me :)

Reply:
Fixed style issues.

Agree. In theory we can never put peer into down state. Removed return

Also put return nil into initialization of this key into spawn_checker()